data_steward/cdr_cleaner/cleaning_rules/remove_participant_data_past_deactivation_date.py

```diff
@@ -29,85 +29,112 @@
 
     def get_query_specs(self):
         """
-        This function generates a list of query dicts for ensuring the dates and datetimes are consistent
+        This function generates a list of query dicts.
 
-        :return: a list of query dicts for ensuring the dates and datetimes are consistent
-        """
+        These queries should sandbox and remove all data past the
+        participant's deactivation date.
 
-        deactivation_queries = remove_ehr_data_queries(
-            bq.get_client(self.project_id), self.api_project_id,
-            self.project_id, self.dataset_id, self.sandbox_dataset_id)
-        return deactivation_queries
+        :return: a list of query dicts
+        """
+        deact_table_ref = gbq.TableReference.from_string(self.destination_table)
+        # creates sandbox and truncate queries to run for deactivated participant data drops
+        # setup_rule must be run before this to ensure the client is properly
+        # configured.
+        queries = rdp.generate_queries(self.client,
+                                       self.project_id,
+                                       self.dataset_id,
+                                       self.sandbox_dataset_id,
+                                       deact_table_ref,
+                                       data_stage_id=self.table_namer)
+        return queries
 
     def setup_rule(self, client):
         """
-        Function to run any data upload options before executing a query.
+        Responsible for grabbing and storing deactivated participant data.
+
+        :param client: client object passed to store the data
         """
-        pass
+        LOGGER.info("Querying RDR API for deactivated participant data")
+        # gets the deactivated participant dataset to ensure it's up-to-date
+        df = psr.get_deactivated_participants(self.api_project_id,
+                                              DEACTIVATED_PARTICIPANTS_COLUMNS)
+
+        LOGGER.info(f"Found '{len(df)}' deactivated participants via RDR API")
+
+        # To store dataframe in a BQ dataset table named _deactivated_participants
+        psr.store_participant_data(df, self.project_id, self.destination_table)
+
+        LOGGER.info(f"Finished stored participant records in: "
+                    f"`{self.destination_table}`")
+
+        LOGGER.debug("instantiating class client object")
+        self.client = client
 
     def get_sandbox_tablenames(self):
         """
-        Returns an empty list because this rule does not use sandbox tables.
+        Return a list table names created to backup deleted data.
         """
-        return []
+        LOGGER.info("Generating static list of possible sandbox table names")
+
+        if self.client:
+            LOGGER.info(f"Getting live table names from: `{self.dataset_id}`")
+            tables_list = self.client.list_tables(self.dataset_id)
+
+            table_names = [table_item.table_id for table_item in tables_list]
```

Comment:
In this case `self.affected_tables = common.CDM_TABLES + common.FITBIT_TABLES`.  The current implementation is consistent with the original implementation.  It does not assume that `self.affected_tables` is the source of truth for this rule.  When a client is available, it assumes the live dataset is the source of truth.  Since we are trying to make sure to remove all post deactivation data, I think it makes more sense to use the ultimate source of truth.  

But maybe I could make a change to the value of `self.affected_tables` when `setup_rule` is run to make this better.  I'll try it.